src/PrestaShopBundle/Controller/Admin/Improve/Design/CmsPageController.php

```
@@ -0,0 +1,378 @@
+{
+    /**
+     * Responsible for displaying page content.
+     *
+     * @AdminSecurity("is_granted('read', request.get('_legacy_controller'))")
+     *
+     * @Template("@PrestaShop/Admin/Improve/Design/Cms/index.html.twig")
+     *
+     * @param int $cmsCategoryParentId
+     * @param CmsPageCategoryFilters $filters
+     *
+     * @return array
+     */
+    public function indexAction($cmsCategoryParentId, CmsPageCategoryFilters $filters)
+    {
+        $cmsCategoryGridFactory = $this->get('prestashop.core.grid.factory.cms_page_category');
+        $cmsCategoryGrid = $cmsCategoryGridFactory->getGrid($filters);
+
+        $gridPresenter = $this->get('prestashop.core.grid.presenter.grid_presenter');
+
+        $viewData = $this->get('prestashop.core.cms_page.data_provider.cms_page_view')->getView($cmsCategoryParentId);
+
+        return [
+            'cmsCategoryGrid' => $gridPresenter->present($cmsCategoryGrid),
+            'cmsPageView' => $viewData,
+        ];
+    }
+
+    /**
+     * Implements filtering for the cms page category list.
+     *
+     * @param int $cmsCategoryParentId
+     * @param Request $request
+     *
+     * @return RedirectResponse
+     */
+    public function searchCategoryAction($cmsCategoryParentId, Request $request)
+    {
+        $definitionFactory = $this->get('prestashop.core.grid.definition.factory.cms_page_category');
+        $definitionFactory = $definitionFactory->getDefinition();
+
+        $gridFilterFormFactory = $this->get('prestashop.core.grid.filter.form_factory');
+        $searchParametersForm = $gridFilterFormFactory->create($definitionFactory);
+        $searchParametersForm->handleRequest($request);
+
+        $filters = [];
+        if ($searchParametersForm->isSubmitted()) {
+            $filters = $searchParametersForm->getData();
+        }
+
+        return $this->redirectToRoute('admin_cms_pages_index', compact('cmsCategoryParentId', 'filters'));
```

even though `compact` seems like a cool thing to use, i would not recommend doing so, mainly because if you were to change `$cmsCategoryParentId` variable name and forget to update `compact` params, you wont get any error as it silently skips parameters that are not defined, so it may hide bugs that are really hard to track down.

second, it forces you to make array keys `camelCase` which i think is against Symfony's coding standard (but i may be wrong regarding that).

@matks wdyt about it? personally, i dont want to encourage use of `compact` in core. :/